app/src/frontend/building/data-container.tsx

```diff
@@ -80,11 +81,18 @@ const withCopyEdit: (wc: React.ComponentType<CategoryViewProps>) => DataContaine
         static getDerivedStateFromProps(props, state): DataContainerState {
             const newBuildingId = props.building == undefined ? undefined : props.building.building_id;
             const newBuildingRevisionId = props.building == undefined ? undefined : props.building.revision_id;
+            const categoryKeys = {};
+
+            for (let key in dataFields) {
+                if (dataFields[key].category == props.cat && props.building[key] != null){
+                    categoryKeys[key] = true;
+                }
+            }
```

Comment:
Hi @tomalrussell I've implemented your idea for filtering on category and only selecting fields that are filled within `getDerivedStateFromProps`

The part that doesn't work currently is `props.building[key]`. The props.building allows me to use a key that comes from the Building interface in `frontend/models/building.ts` e.g. `revision_id`, but not a key from `frontend/config/data-fields-config.ts` e.g. `date_year`, which I would have expected to be the case. When I `console.log(props.building)` it looks like a dictionary structure where all the keys are at the same level. Any thoughts?

Reply:
Ok think I've got a workaround for this actually